databases/functions/users/find-signed-user-id.js

```diff
@@ -1,13 +1,27 @@
 const CryptoUtils = require('../../../utils/crypto');
 
-module.exports = async (userModel, userId) => {
+module.exports = async (userModel, userId, options = {}) => {
+  const {shouldReturnObject = false} = options;
+
   const user = await userModel.findOne({
     where: {
       user_id: userId
     },
     raw: true
   });
 
+  if (shouldReturnObject && user?.is_anonymous) {
+    return {
+      user_id: CryptoUtils.decryptAnonymousUserId(user?.encrypted),
+      is_user_anonymous: user?.is_anonymous
+    };
+  } else if (shouldReturnObject) {
+    return {
+      user_id: user?.user_id,
+      is_user_anonymous: user?.is_anonymous
+    };
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `find-signed-user-id.js` now has an optional parameter `options` which defaults to an empty object. This is a good practice for optional parameters. However, the function now has two different return types: it can either return an object or a single value (user_id). This could lead to confusion and potential bugs in the future as the caller needs to be aware of this behavior and handle both cases. It would be better to always return an object for consistency.

```diff
-  if (shouldReturnObject && user?.is_anonymous) {
-    return {
-      user_id: CryptoUtils.decryptAnonymousUserId(user?.encrypted),
-      is_user_anonymous: user?.is_anonymous
-    };
-  } else if (shouldReturnObject) {
-    return {
-      user_id: user?.user_id,
-      is_user_anonymous: user?.is_anonymous
-    };
-  }
-  if (user?.is_anonymous) return CryptoUtils.decryptAnonymousUserId(user?.encrypted);
-  return user?.user_id;

+  return {
+    user_id: user?.is_anonymous ? CryptoUtils.decryptAnonymousUserId(user?.encrypted) : user?.user_id,
+    is_user_anonymous: user?.is_anonymous
+  };
```
This change ensures that the function always returns an object with the same structure, making it easier to use and understand.


<!-- This is an auto-generated comment by OSS CodeRabbit -->